x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -33,30 +32,30 @@ func (k Keeper) SettlePendingClaims(ctx cosmostypes.Context) (
 ) {
 	logger := k.Logger().With("method", "SettlePendingClaims")
 
-	expiringClaims, err := k.GetExpiringClaims(ctx)
-	if err != nil {
-		return settledResults, expiredResults, err
-	}
-
-	// Capture the applications initial stake which will be used to calculate the
-	// max share any claim could burn from the application stake.
-	// This ensures that each supplier can calculate the maximum amount it can take
-	// from an application's stake.
-	applicationInitialStakeMap, err := k.getApplicationInitialStakeMap(ctx, expiringClaims)
-	if err != nil {
-		return settledResults, expiredResults, err
-	}
+	// Use an iterator to get all claims that are expiring at the current block height
+	// instead of getting them all at once to improve scalability.
```

Comment:
@red-0ne This comment ONLY makes sense in the context of this PR. No one other than me is going to read it.

Need to make sure it's "evergreen."

```suggestion
	// Retrieve all expiring claims as an iterator.
	// DEV_NOTE: This previously retrieve a list but has been change to account for large claim counts.
```